backend/src/api/paths/problems.ts

```diff
@@ -194,28 +226,96 @@ const getSubmissionsByProblemIdRoute = createRoute({
 })
 
 // ルートの設定
-app.openapi(getProblemsRoute, (c) => {
-  // TODO: 実際のデータベースクエリを実装
-  const problems: z.infer<typeof schemas.Problem>[] = [
-    {
-      body: "この問題の本文です。",
-      id: 1,
-      supported_languages: [{ name: "Python", version: "3.9" }],
-      test_cases: [{ input: "入力例", output: "出力例" }],
-      title: "サンプル問題",
+app.openapi(getProblemsRoute, async (c) => {
+  const problems = await prisma.problem.findMany({
+    include: {
+      supportedLanguages: {
+        include: {
+          language: true,
+        },
+      },
+      testCases: true,
     },
-  ]
-  return c.json(problems)
+  })
+
+  const formattedProblems = problems.map((problem: ProblemWithRelations) => ({
```

Comment:
IMO: `problem`の型は推論できるので、明示的に書かなくて良いかなと思いました。併せて、上の方で定義され`ているProblemWithRelation`型についても、推論できる個所の型注釈でのみ使われているため削除しても良いかなと思いました

```suggestion
  const formattedProblems = problems.map((problem) => ({
```